src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -265,22 +265,27 @@ class WorkspaceService(protected val userInfo: UserInfo, dataSource: DataSource,
     )
   }
 
-  def cloneWorkspace(sourceWorkspaceName: WorkspaceName, destWorkspaceName: WorkspaceName): Future[PerRequestMessage] =
-    dataSource.inFutureTransaction(readLocks=Set(sourceWorkspaceName), writeLocks=Set(destWorkspaceName)) { txn =>
+  def cloneWorkspace(sourceWorkspaceName: WorkspaceName, destWorkspaceRequest: WorkspaceRequest): Future[PerRequestMessage] =
+    dataSource.inFutureTransaction(readLocks=Set(sourceWorkspaceName), writeLocks=Set(destWorkspaceRequest.toWorkspaceName)) { txn =>
       withWorkspaceContextAndPermissions(sourceWorkspaceName,WorkspaceAccessLevels.Read,txn) { sourceWorkspaceContext =>
-        withNewWorkspaceContext(
-          WorkspaceRequest(
-            namespace = destWorkspaceName.namespace,
-            name = destWorkspaceName.name,
-            realm = sourceWorkspaceContext.workspace.realm,
-            attributes = sourceWorkspaceContext.workspace.attributes),txn) { destWorkspaceContext =>
+
+        // add to or replace current attributes, on an individual basis
+        val newAttrs = sourceWorkspaceContext.workspace.attributes ++ destWorkspaceRequest.attributes
```

Comment:
This was not explicitly asked for, but a WorkspaceRequest comes with attributes, and this seemed like the most reasonable way to handle them.  I added tests for this behavior as well.

The alternative would be adding a new case class for clone workspace requests only with namespace, name, and realm.